lib/variable/test/to_unit_test.cpp

```diff
@@ -274,3 +274,19 @@ TEST(ToUnitTest, small_to_large_rounding_error_double) {
                     units::Unit("m")),
             one_m);
 }
+
+TEST(ToUnitTest, small_number_to_small_unit) {
+  const auto area = units::angstrom * units::angstrom;
+  const auto small = makeVariable<double>(units::Unit("m**2"), Values{1e-20});
+  const auto result = to_unit(small, area);
+  EXPECT_EQ(result.unit(), area);
+  EXPECT_DOUBLE_EQ(result.value<double>(), 1.0);
+}
```

Comment:
Did you check if this test passes without going through `base_scale`? I'm wondering how much if anything we are gaining from rounding.

Reply:
Well, probably that is two different questions:
- Even with a scale of 1e9 we gain by going through `base_scale`.
- This test in particular is essentially a safeguard against massive issues, not rounding errors, so I don't know if `EXPECT_DOUBLE_EQ` is sensitive to those rounding errors.

Reply:
But to answer your actual question: I don't think rounding does anything in many of these cases. But this seemed to be the solution with the simplest branching logic (remember our chat, where I had temporarily tried putting a limit on `scale`).